### PR TITLE
Autotest: copter: add test for mandatory alt estimation for Alt_hold

### PR DESCRIPTION
**Pull request by khancyr**
Implement test for https://github.com/ArduPilot/ardupilot/pull/15092

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -5148,6 +5148,31 @@ def test_parameter_validation(self):
...
+        self.context_push()
+        ex = None
+        try:
+            self.set_parameter("EK2_ALT_SOURCE", 2)
+            self.wait_gps_disable()
+            self.wait_heartbeat()
+            self.change_mode("ALT_HOLD")
+            self.assert_prearm_failure("Need Alt Estimate")
+            self.change_mode("STABILIZE")
+            self.zero_throttle()
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:5161]: Shouldn't need to zero throttle rc input.  If you do the test suite has stuffed up somehow - it's one of the guarantees we make.

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -5148,6 +5148,31 @@ def test_parameter_validation(self):
         self.drain_mav()
         self.assert_prearm_failure("Check MOT_PWM_MIN/MAX")
 
+    def test_alt_estimate_prearm(self):
+        self.context_push()
+        ex = None
+        try:
+            self.set_parameter("EK2_ALT_SOURCE", 2)
+            self.wait_gps_disable()
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:5156]: We should change this to not be of the "wait_" signature, which don't typically make changes, just wait for something.

Not this PR, 'though.

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -5148,6 +5148,31 @@ def test_parameter_validation(self):
...
+            self.wait_heartbeat()
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:5157]: This wait_heartbeat() should be in `change_mode()`.

File: Tools/autotest/arducopter.py [hunk 4]
```diff
@@ -5148,6 +5148,31 @@ def test_parameter_validation(self):
...
+            self.change_mode("ALT_HOLD")
+            self.assert_prearm_failure("Need Alt Estimate")
+            self.change_mode("STABILIZE")
+            self.zero_throttle()
+            self.arm_vehicle()
+            self.takeoff()
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py:5163]: You can get takeoff() to arm you and takeoff in stabilize.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Cool.

File: Tools/autotest/arducopter.py (hunk 4 above)
Comment (khancyr) [reply Tools/autotest/arducopter.py:5163]: yes but I don't want the wait_EKF_ready ;-P ! 
I just copied the 3 previous lines from the takeoff() function

File: Tools/autotest/arducopter.py (hunk 2 above)
Comment (khancyr) [reply Tools/autotest/arducopter.py:5156]: agree! we should a set and a wait function

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/arducopter.py:5161]: done

File: Tools/autotest/arducopter.py (hunk 3 above)
Comment (khancyr) [reply Tools/autotest/arducopter.py:5157]: done

**Comment by khancyr** [comment]
address @peterbarker comments

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**